clang/lib/Headers/hlsl/hlsl_intrinsics.h

```diff
@@ -570,7 +716,13 @@ double4 sin(double4);
 //===----------------------------------------------------------------------===//
 // sqrt builtins
 //===----------------------------------------------------------------------===//
+
+/// \fn T sqrt(T Val)
+/// \brief Returns the square root of the input value, \a Val.
+/// \param Val The input value.
+
 #ifdef __HLSL_ENABLE_16_BIT
+_HLSL_AVAILABILITY(shadermodel, 6.2)
```

Comment:
Surprisingly, it looks like there aren't function prototypes for half2/3/4, and I imagine we would want this new attribute applied to such aliases. Just a note for the future.

Reply:
That's a good catch. I think we're missing all the vector overloads for `sqrt`. I also have another fix coming that specifically adjusts the `half` handling.